updating the docstring for full_like function

someshfengde opened the pull request.

someshfengde commented:
#2120

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -285,13 +285,66 @@ def full_like(
+        >>> y = ivy.full_like(x, fill_value)
+        >>> print(y)
+        ivy.array([0.001, 0.001, 0.001, 0.001, 0.001, 0.001])
+
```
please add ivy.Array examples

someshfengde replied:
I'll add more ivy.array examples

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -285,13 +285,66 @@ def full_like(
+        >>> print(y)
+        ivy.array([[0.000123, 0.000123, 0.000123],
+       [0.000123, 0.000123, 0.000123]])
+
```
please add ivy.Container examples

someshfengde replied:
Hi @juliagsy  I've tried to do full_like function with ivy.Container but it's not working there is issue with function wrapping ig 

I've posted the error log here https://github.com/unifyai/ivy/issues/2120#issuecomment-1191468184

juliagsy replied:
oh! I'll look into them and let you know! Thank you!

someshfengde replied:
got it 

for now I'll add more examples with ivy.array :smile:

juliagsy replied:
Hey @someshfengde I've just pushed a fix, you should be able to add container examples after pulling the latest commit! Thanks!

someshfengde replied:
Hi I am still getting same error

someshfengde replied:
@juliagsy

juliagsy replied:
Hi, have you fetched and merged the latest commit from master?

someshfengde replied:
thanks I got it I need to restart the docker for changes to be applied overall repo 

I'll update the changes with `ivy.Container` now

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -285,13 +285,66 @@ def full_like(
+        >>> y = x.full_like(fill_value)
+        >>> print(y)
+        ivy.array([1, 1, 1, 1, 1, 1])
+
```
please add ivy.Container instance method examples

someshfengde replied:
Hi @juliagsy I've tried to do full_like function with ivy.Container but it's not working there is issue with function wrapping ig

I've posted the error log here https://github.com/unifyai/ivy/issues/2120#issuecomment-1191468184

someshfengde replied:
got it

juliagsy reviewed (changes requested):
Hello! Great work so far! Some additional examples are needed as stated in the comments below. Moreover, please also add some examples to the `full_like` method in the `ivy/array/creation.py` (ivy.array examples) and `ivy/container/creation.py` (ivy.container and mixture of ivy.array and ivy.container examples) files. Once you're done, please request my review again! Thanks for contributing!

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,67 @@ def full_like(
+            >>> import ivy
+            >>> x = ivy.array([1.0, 2.0, 3.0, 4.0, 5.0, 6.0]) # for filling values with floating datatype the array is required to be of same dtype (float in this case)
+            >>> fill_value = 0.000123
+            >>> y = ivy.full_like(x, fill_value)
```
please use `x.full_like(fill_value)` to showcase instance method

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,67 @@ def full_like(
+            >>> print(y)
+            ivy.array([0.000123, 0.000123, 0.000123, 0.000123, 0.000123, 0.000123])
+
+        with torch.Tensor input:
```
please remove this example as it is part of native array examples

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,67 @@ def full_like(
+            ivy.array([[0.000123, 0.000123, 0.000123],
+           [0.000123, 0.000123, 0.000123]])
+
+        Instance Method Examples:
```
you can change the title above to instance method examples and remove this part as they will be identical

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -300,13 +300,66 @@ def full_like(
+        >>> print(y)
+        ivy.array([0.000123, 0.000123, 0.000123, 0.000123, 0.000123, 0.000123])
+
+    with torch.Tensor input:
```
this is not needed as torch tensors are part of native array examples, it does not need to have examples on its own

someshfengde commented:
Hi I've made more changes 
I am not sure what can I add into the `ivy/container/creation.py` can you please suggest what things should I add 
for now I've just added the same things about containers as I added into `ivy/array/cration.py`

someshfengde commented:
hi @juliagsy  full_like function is not implemented in `mxnet/creation.py` I think it's supposed to be there cause full_like function is present in all other backends (torch,tensorflow, numpy and jax)

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,50 @@ def full_like(
+            >>> print(y)
+            ivy.array([1, 1, 1, 1, 1, 1])
+
+        With ivy.Container input
```
we don't need container examples in array instance methods

someshfengde replied:
ohh got it

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,50 @@ def full_like(
+        With int datatype:
+            >>> x = ivy.array([1,2,3])
+            >>> fill_value = 0
+            >>> ivy.full_like(x , fill_value)
```
you may change this to `x.full_like(fill_value)`  as we don't need functional examples here

juliagsy replied:
then replace the title with `Instance Method Examples` from below

juliagsy commented on ivy/container/creation.py:
```diff
@@ -157,6 +157,26 @@ def full_like(
         dtype: Optional[Union[ivy.Dtype, ivy.NativeDtype]] = None,
         device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
     ) -> ivy.Container:
+        """
```
You may refer to the docstrings formatting here, please also add docstrings for parameter and return. You can use example types there as reference too!

juliagsy replied:
Please reformat for both `full_like` and `static_full_like`

juliagsy replied:
parameters description can be taken from functional `full_like` and description from key_chains onwards can be taken from the reference too!

juliagsy reviewed (changes requested):
Great work so far! I've provided reference for the ivy.Container examples in one of the related comments! You're almost there, please request my review once you're done, thanks for your hard work!

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -300,13 +300,88 @@ def full_like(
+        >>> print(y)
+        ivy.array([0.000123, 0.000123, 0.000123, 0.000123, 0.000123, 0.000123])
+
+    With torch.Tensor input:
```
one last comment, please remove this torch.Tensor examples as it is a subcategory of ivy.native_array, we have these in the next section and that's sufficient! Thanks

someshfengde replied:
done ;)

someshfengde commented:
I've made those changes let me know if there are any other changes required :smile:

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,31 @@ def full_like(
+        -------------------
+
+        With int datatype:
+            >>> x = ivy.array([1,2,3])
```
just to keep them formatting consistent, would you mind making a small change so that there's a blank line between subtitle and example code, then example code aligned with title, eg.

```
With int datatype:

>>> x = ivy.array([1,2,3])
...
```

juliagsy replied:
Please follow the convention for all examples, thanks!

juliagsy commented on ivy/array/creation.py:
```diff
@@ -79,6 +79,31 @@ def full_like(
             to ``fill_value``.
 
+        Instance Method Examples:
+        -------------------
```
more dashes to align with title

juliagsy commented on ivy/container/creation.py:
```diff
@@ -132,6 +132,63 @@ def static_full_like(
+
+        Examples
+        --------
+       With ivy.Container input
```
alignment and formatting as explained above

juliagsy commented on ivy/container/creation.py:
```diff
@@ -157,6 +214,63 @@ def full_like(
         device: Optional[Union[ivy.Device, ivy.NativeDevice]] = None,
     ) -> ivy.Container:
+        """
+        ivy.Container static method variant of ivy.full_like. This method simply wraps the
```
`instance method` instead of `static...`

juliagsy commented on ivy/container/creation.py:
```diff
@@ -157,6 +214,63 @@ def full_like(
+
+        Examples
+        --------
+        With ivy.Container input
```
alignment and formatting as explained above

juliagsy commented on ivy/functional/ivy/creation.py:
```diff
@@ -300,13 +300,77 @@ def full_like(
-    ivy.array([1, 1, 1, 1, 1, 1])
+    Functional Examples
+    -------------------
+    With int datatype:
```
alignment and formatting as explained above

juliagsy reviewed (changes requested):
No worries at all! Thanks for making all the changes too! 

The examples look good so far! The last thing now is just alignment and reformatting so that the docstrings are consistent across functions and modules, there're stated in the comments below!

As usual, please request my review once you're done! Thanks a lot for all the hard work!

someshfengde commented:
updated the formatting for docstrings

someshfengde commented:
the zeros_like function was having this kind of docstring I took reference from that function that's why I did this kinda formatting

juliagsy commented:
ahh I see, no worries! btw, I'll go over a last check asap and as usual, will either request a change, or merge it if it's good to go! Thanks for contributing!!

juliagsy reviewed (approved):
Thanks for contributing!

someshfengde commented:
thank you @juliagsy :smile:
